src/main/java/functions/Parser.java

```diff
@@ -0,0 +1,223 @@
+                    break;
+
+                case EVENT:
+                    eventCommandHandler(taskList, input, secondHalfInput);
+                    break;
+
+                case FIND:
+                    findCommandHandler(taskList, secondHalfInput);
+
+                default:
+                    break;
+            }
+        } catch (IllegalArgumentException e) {
+            // If task inserted not an ENUM
+            System.out.println("Oops!!! I'm sorry but I don't know what that means :-(");
+            System.out.println("Please use one of the following commands: list, mark, unmark," +
+                    " delete, todo, deadline, event, bye");
+        }
+    }
+
+    public void listCommandHandler(TaskList taskList) {
+        for (int i=0; i<taskList.size(); i++) {
+            String message = String.format("%d. %s", i+1, taskList.get(i).getTaskAsString());
+            System.out.println(message);
+        };
+    }
+
+    public void markCommandHandler(String[] inputArray, TaskList taskList) {
+        try {
+            int targetTaskIdx = Integer.parseInt(inputArray[1]) -1;
+            Task task = taskList.get(targetTaskIdx);
+            task.markAsDone();
+            System.out.println("Nice! I've marked this task as done:");
+            System.out.println(task.getTaskAsString());
+        } catch (NumberFormatException e) {
+            System.out.println("Invalid number provided. Please provide in the form of 'mark {task number}'. Eg: 'mark 1' to mark task 1.");
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("Invalid task number provided. Please provide in the form of 'mark {task number}'. Eg: 'mark 1' to mark task 1.");
+        }
+    }
+
+    public void unmarkCommandHandler(String[] inputArray, TaskList taskList) {
+        try {
+            int targetTaskIdx = Integer.parseInt(inputArray[1]) -1;
+            Task task = taskList.get(targetTaskIdx);
+            task.markAsUndone();
+            System.out.println("Ok! I've marked this task as not done yet:");
+            System.out.println(task.getTaskAsString());
+        } catch (NumberFormatException e) {
+            System.out.println("Invalid number provided. Please provide in the form of 'unmark {task number}'. Eg: 'unmark 1' to unmark task 1.");
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("Invalid task number provided. Please provide in the form of 'unmark {task number}'. Eg: 'unmark 1' to unmark task 1.");
+        }
+    }
+
+    public void deleteCommandHandler(String[] inputArray, TaskList taskList) {
+        try {
+            int targetTaskIdx = Integer.parseInt(inputArray[1]) -1;
+            Task task = taskList.get(targetTaskIdx);
+            taskList.remove(targetTaskIdx);
+            System.out.println("Noted: I've removed this task:");
+            System.out.println(task.getTaskAsString());
```

Comment:
Maybe each logical units within a block should be separated by one blank line?

```suggestion
            // Get the task and task index
            int targetTaskIdx = Integer.parseInt(inputArray[1]) -1;
            Task task = taskList.get(targetTaskIdx);
            
            taskList.remove(targetTaskIdx);
            
            System.out.println("Noted: I've removed this task:");
            System.out.println(task.getTaskAsString());
```

Rationale: Enhances readability by introducing white space between logical units. Each block is often introduced by a comment.